video-app/src/app/library/add/add.component.ts

```diff
@@ -34,25 +41,49 @@ export class AddComponent {
   constructor(private readonly router: Router
     ,         private readonly youtubeService: YoutubeService
     ,         private readonly vimeoService: VimeoService
-    ,         private readonly streamingPlatform: StreamingPlatformService) { }
+    ,         private readonly streamingPlatform: StreamingPlatformService
+    ,         private readonly store: Store<{ videos: VideoState }>) {
+      this.video$ = store.pipe(select('videos'));
+     }
+
+    public ngOnInit(): void {
+      this.videoSubscription = this.video$
+        .pipe(
+          map(x => {
```

Comment:
`x` to mało mówiąca nazwa ;) 

dodatkowo ważna sprawa przy subskrypcjach - https://netbasal.com/why-its-important-to-unsubscribe-from-rxjs-subscription-a7a6455d6a02